contracts/FlowTransactionSchedulerUtils.cdc

```diff
@@ -579,6 +582,269 @@ access(all) contract FlowTransactionSchedulerUtils {
+                    coaEVMAddress: self.coaCapability.borrow()?.address(),
+                )
+            }
+            if viewType == Type<StoragePath>() {
+                return FlowTransactionSchedulerUtils.coaHandlerStoragePath()
+            } else if viewType == Type<PublicPath>() {
+                return FlowTransactionSchedulerUtils.coaHandlerPublicPath()
+            } else if viewType == Type<MetadataViews.Display>() {
+                return MetadataViews.Display(
+                    name: "COA Scheduled Transaction Handler",
+                    description: "Scheduled Transaction Handler that can execute transactions on behalf of a COA",
+                    thumbnail: MetadataViews.HTTPFile(
+                        url: ""
+                    )
+                )
+            }
+            return nil
+        }
+    }
+
+    /// Enum for COA handler execution type
+    access(all) enum COAHandlerTxType: UInt8 {
+        access(all) case DepositFLOW
+        access(all) case WithdrawFLOW
+        access(all) case Call
+
+        // TODO: Should we have other transaction types??
+    }
+
+    access(all) struct COAHandlerParams {
+
+        /// The type of transaction to execute
+        access(all) let txType: COAHandlerTxType
+
+        /// Indicates if the whole set of scheduled transactions should be reverted
+        /// if this one transaction fails to execute in EVM
+        access(all) let revertOnFailure: Bool
+
+        /// The amount of FLOW to deposit or withdraw
+        /// Not required for the Call transaction type
+        access(all) let amount: UFix64?
+
+        /// The following fields are only required for the Call transaction type
+        access(all) let callToEVMAddress: EVM.EVMAddress?
+        access(all) let data: [UInt8]?
+        access(all) let gasLimit: UInt64?
+        access(all) let value: EVM.Balance?
+
+        init(txType: UInt8, revertOnFailure: Bool, amount: UFix64?, callToEVMAddress: String?, data: [UInt8]?, gasLimit: UInt64?, value: UInt?) {
+            self.txType = COAHandlerTxType(rawValue: txType)
+                ?? panic("Invalid COA transaction type enum")
+            self.revertOnFailure = revertOnFailure
+            if self.txType == COAHandlerTxType.DepositFLOW {
+                assert(amount != nil, message: "Amount is required for deposit but was not provided")
+            }
+            if self.txType == COAHandlerTxType.WithdrawFLOW {
+                assert(amount != nil, message: "Amount is required for withdrawal but was not provided")
+            }
+            if self.txType == COAHandlerTxType.Call {
+                assert(callToEVMAddress != nil && callToEVMAddress!.length == 40, message: "Call to EVM address is required for EVM call but was not provided or is invalid length (must be 40 hex chars)")
+                assert(data != nil, message: "Data is required for EVM call but was not provided")
```

Comment:
Technically, for a `call`, you need either `data` for smart contract calls, or `value` for transfers between accounts. It's also possible that you can have both, but it's not mandatory.